### PR TITLE
Prevent decode to URI param a string with a "+" character

### PR DESCRIPTION
**Pull request by folektoras**
During searching in tweakwise a string with 2 words (for example Big Ball) tweakwise will generate a URL in magento "..catalogsearch/result/?q=Big+Ball'.
Later on on result page if someone tries to apply a new filter, the function _getQParam is reading the URL to check if there is anything after q.
Therefore the variable matches[2] will include a string like "Big+Ball" because is fetched from the URL. When later on, the same function will try to decode this string will generate an incorrect URL with searching the "Big+Ball" instead of "Big Ball". New Url would be '../?q=Big%2BBall'.

In this commit a change is being done in the function _getQParam. This change will first check if string that the user is actually searching for something that contains a "+" or not (always possible someone wants to search for 'Big+Ball').
If the case is that the user is just searching for 'Big Ball' then this string needs to be decoded to a param, generating a correct url.

### CODE REVIEW

**Comment by folektoras** [comment]
@edwinljacobs hello! Could you take a look at this one?

File: src/view/frontend/web/js/navigation-form.js [hunk 1]
```diff
@@ -56,6 +56,13 @@ define(['jquery', 'jquery/ui'], function($) {
         _getQParam: function() {
             let matches = window.location.search.match(/(\?|&)q\=([^&]*)/);
             if (matches && matches[2]) {
+                let trimmedMatch = matches[2].replace(/\+/g, ' '),
+                    searchVal    = jQuery('#search').val();
+
```
Comment (edwinljacobs) [inline src/view/frontend/web/js/navigation-form.js:61]: I would like to discuss this with a collegue of mine, it seems to me there are better ways to get the search field value then to assume its id here. Perhaps, but im not sure, we could ask it from the quickSearch component?

Comment (folektoras) [reply src/view/frontend/web/js/navigation-form.js:61]: @edwinljacobs that would be ideal. Unfortunately I did not have the time to dig really deep into this but the idea would be the same :)

Comment (edwinljacobs) [reply src/view/frontend/web/js/navigation-form.js:61]: I tried to "include" the quicksearch component in the navigation-form.js but failed to do so correctly. Problably my lack of knowledge on requirejs .... ill check if its possible, if not then we will likely merge as it stands :) If you happen to find the answer then please let us know!

Comment (folektoras) [reply src/view/frontend/web/js/navigation-form.js:61]: Alright. If something comes up I will let you know. Thanks :)

Comment (edwinljacobs) [reply src/view/frontend/web/js/navigation-form.js:61]: Merged!

Comment (edwinljacobs) [reply src/view/frontend/web/js/navigation-form.js:61]: This will be included in the next release

**Pull request merged**